PokerGame/PokerEngine/Poker.cs

```diff
@@ -28,17 +34,79 @@ public static List<Card> CalculateWinningHand(List<Card> handOne, List<Card> han
             return CalculateHandWithHighestCardAndKicker(handOne, handTwo);
         }
 
+        private static List<Card> CalculateHandHasWithTheBestStraight(List<Card> handOne, List<Card> handTwo)
+        {
+            Rank rankOfHandOneHighCard;
+            var handOneHasStraight = hasStraight(handOne, out rankOfHandOneHighCard);
+
+            Rank rankOfHandTwoHighCard;
+            var handTwoHasStraight = hasStraight(handTwo, out rankOfHandTwoHighCard);
+
+            if (handOneHasStraight && handTwoHasStraight)
+            {
+                return handWinsWhenEqualRankButHigherCard(handOne, handTwo, rankOfHandOneHighCard, rankOfHandTwoHighCard);
+            }
+
+            if (handOneHasStraight)
+            {
+                return handOne;
+            }
+            if (handTwoHasStraight)
+            {
+                return handTwo;
+            }
+            return null;
+        }
+
+        private static bool hasStraight(List<Card> hand, out Rank rankOfHighestCardInStraight)
+        {
+            var handOrderedByHighestCard = hand
+                .OrderBy(card => card.Rank)
+                .ToList();
+
+            if(handOrderedByHighestCard[handOrderedByHighestCard.Count()-1].Rank == Rank.Ace)
+            {
+                handOrderedByHighestCard.RemoveAt(handOrderedByHighestCard.Count - 1);
```

Comment:
Why remove the ace?  I can't understand this.  Oh, and a better way to get the last item in a collection is using ```handOrderedByHighestCard.Last()```